projects/core/src/site-context/site-context.module.ts

```diff
@@ -62,6 +86,18 @@ export class SiteContextModule {
           ],
           multi: true,
         },
+        {
+          provide: APP_INITIALIZER,
+          useFactory: languageStatePersistenceFactory,
+          deps: [LanguageStatePersistenceService, ConfigInitializerService],
+          multi: true,
+        },
+        {
+          provide: APP_INITIALIZER,
+          useFactory: currencyStatePersistenceFactory,
+          deps: [CurrencyStatePersistenceService, ConfigInitializerService],
+          multi: true,
+        },
```

Comment:
This module is busy with providers. What about moving them (and the factory functions) to a separate file and here only spread them? For instance `...contextPersistenceProviders`?